Entities/ACrab.cpp

```diff
@@ -531,33 +483,122 @@ Vector ACrab::GetCPUPos() const
 
 Vector ACrab::GetEyePos() const
 {
-    if (m_pTurret && m_pTurret->IsAttached() && m_pTurret->GetMountedMO())
-        return m_pTurret->GetMountedMO()->GetPos();
+    if (m_pTurret && m_pTurret->IsAttached() && m_pTurret->HasMountedDevice())
+        return m_pTurret->GetMountedDevice()->GetPos();
 
     return m_Pos;
 }
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-//////////////////////////////////////////////////////////////////////////////////////////
-// Virtual method:  SetID
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Sets the MOID of this MovableObject for this frame.
+Attachable *ACrab::GetTurret() const {
+    return (m_pTurret && m_pTurret->IsAttached()) ? dynamic_cast<Attachable *>(m_pTurret) : nullptr;
```

Comment:
Yeah, this was due to the templated methods I had before, they all had to only use Attachable because otherwise it wouldn't work. Changing this in the getters is straightforward and should work fine, but it's tougher in the setters.
The setters generally look like this
```
void Class::SetThing(Attachable newThing) {
	if (newThing == nullptr) {
		Remove old thing and clear thing pointer
	} else {
		Cast newThing to appropriate type so we're not trying to put garbage in. This isn't great, I agree
		Remove old thing
		Add newThing as attachable and set thing pointer
		Set up lambda function and save it in the map of hardcoded attachable setters. This lambda function MUST have arguments `(MOSRotating *thisObject, Attachable *attachable)` in order to be added to the map of setters, so I can't change this part.
	}
}
```
The current lambdas look like
`setterMap.insert({whatever key, [](MOSRotating *parent, const Attachable *attachable) { dynamic_cast<ParentType *>(parent)->HardcodedSetter(attachable); }});`
In order to make the HardcodedSetter take its specific type here, I'd have to put the cast in the lambda. This is probably actually a good thing, since it'd make the methods safer (i.e. you can only pass in the exact right thing or it won't work), but it won't save a cast, because the lambda would look like
`setterMap.insert({whatever key, [](MOSRotating *parent, const Attachable *attachable) { dynamic_cast<ParentType *>(parent)->HardcodedSetter(dynamic_cast<ProperAttachableSubType *>(attachable)); }});

Reply:
Fair enough about the setters, just minimize the casts wherever it's simple enough to do.
Dynamic casting is expensive y u taek muh gainzz.

Reply:
I'll think a little more on this and maybe change it. I may also be able use static_cast instead of dynamic if I go the route I mentioned above, which would be nice.
Good thing, at least, is that this isn't being called on the regular. Mostly gonna happen when something is made or dies, so it's not the end of the world.

Reply:
Things are made quite frequently and die non-stop, so this is being called on the regular. ¯\\\_(ツ)_/¯

Reply:
Yeah, that's true. I meant every frame really. But yeah, when it's called are the times we need every bit of performance we can get lol

Reply:
I've changed this as mentioned above, so the setters take in the appropriate class and the lambdas do casting. No real change in efficiency I think, but it makes things a bit safer and cleaner. As such, I'm resolving this conversation.